tests/test_multipart.py

```diff
@@ -136,13 +136,34 @@ def test_multipart_encode_files_allows_filenames_as_none():
 
         assert content_type == f"multipart/form-data; boundary={boundary}"
         assert body == (
-            '--{0}\r\nContent-Disposition: form-data; name="file"\r\n'
-            "Content-Type: application/octet-stream\r\n\r\n<file content>\r\n"
-            "--{0}--\r\n"
+            '--{0}\r\nContent-Disposition: form-data; name="file"\r\n\r\n'
+            "<file content>\r\n--{0}--\r\n"
             "".format(boundary).encode("ascii")
         )
 
 
+@pytest.mark.parametrize(
+    "file_name,expected_content_type",
+    [("example.json", "application/json"), ("example.log", "application/octet-stream")],
+)
+def test_multipart_encode_files_guesses_correct_content_type(
+    file_name, expected_content_type
+):
+    files = {"file": (file_name, io.BytesIO(b"<file content>"))}
+    with mock.patch("os.urandom", return_value=os.urandom(16)):
+        boundary = binascii.hexlify(os.urandom(16)).decode("ascii")
+
+        body, content_type = multipart.multipart_encode(data={}, files=files)
+
+        assert content_type == f"multipart/form-data; boundary={boundary}"
+        assert body == (
+            '--{0}\r\nContent-Disposition: form-data; name="file"; filename="{2}"\r\n'
+            "Content-Type: {1}\r\n\r\n<file content>\r\n"
+            "--{0}--\r\n"
+            "".format(boundary, expected_content_type, file_name).encode("ascii")
```

Comment:
Would 3.6+ f-strings be a tad more readable here?

```python
assert body == (
    f'--{boundary}\r\nContent-Disposition: form-data; name="file"; filename="{filename}"\r\n'
    f"Content-Type: {expected_content_type}\r\n\r\n<file content>\r\n"
    f"--{boundary}--\r\n"
).encode("ascii")
```

Reply:
Definitely would. I was trying to mirror the original style of code by copying from a similar test, but this is far better.